File: src/main/java/com/cxxcxx/zinecraft/api/world/dimension/DimensionHelper.java
package com.cxxcxx.zinecraft.api.world.dimension;

import java.util.OptionalLong;

import net.minecraft.tags.BlockTags;
import net.minecraft.util.valueproviders.IntProvider;
import net.minecraft.util.valueproviders.UniformInt;
import net.minecraft.world.level.dimension.BuiltinDimensionTypes;
import net.minecraft.world.level.dimension.DimensionType;
import net.minecraft.world.level.dimension.DimensionType.MonsterSettings;
import org.jetbrains.annotations.NotNull;

public final class DimensionHelper {
  @NotNull
  public static final DimensionHelper INSTANCE = new DimensionHelper();

  private DimensionHelper() {
  }

  @NotNull
  public final DimensionType overworldLikeType() {
    return new DimensionType(
        OptionalLong.empty(),
        true,
        false,
        false,
        true,
        1.0,
        true,
        false,
        -64,
        384,
        384,
        BlockTags.INFINIBURN_OVERWORLD,
        BuiltinDimensionTypes.OVERWORLD_EFFECTS,
        0.0F,
        new MonsterSettings(false, true, (IntProvider) UniformInt.of(0, 7), 0)
    );
  }
}

File: src/main/kotlin/com/cxxcxx/zinecraft/api/world/dimension/DimensionHelper.kt
package com.cxxcxx.zinecraft.api.world.dimension

import net.minecraft.tags.BlockTags
import net.minecraft.util.valueproviders.UniformInt
import net.minecraft.world.level.dimension.BuiltinDimensionTypes
import net.minecraft.world.level.dimension.DimensionType
import java.util.OptionalLong

/** 构造维度动态注册表值时使用的无状态辅助方法。 */
object DimensionHelper {
  fun overworldLikeType(): DimensionType = DimensionType(
    OptionalLong.empty(),
    true,
    false,
    false,
    true,
    1.0,
    true,
    false,
    -64,
    384,
    384,
    BlockTags.INFINIBURN_OVERWORLD,
    BuiltinDimensionTypes.OVERWORLD_EFFECTS,
    0.0f,
    DimensionType.MonsterSettings(false, true, UniformInt.of(0, 7), 0)
  )

}
